synapse/rest/admin/users.py

```diff
@@ -486,12 +486,22 @@ async def on_GET(self, request, user_id):
 class DeactivateAccountRestServlet(RestServlet):
     PATTERNS = admin_patterns("/deactivate/(?P<target_user_id>[^/]*)")
 
-    def __init__(self, hs):
+    def __init__(self, hs: "HomeServer"):
         self._deactivate_account_handler = hs.get_deactivate_account_handler()
         self.auth = hs.get_auth()
+        self.is_mine = hs.is_mine
+        self.store = hs.get_datastore()
 
-    async def on_POST(self, request, target_user_id):
+    async def on_POST(self, request: str, target_user_id: str) -> Tuple[int, JsonDict]:
         await assert_requester_is_admin(self.auth, request)
+
+        if not self.is_mine(UserID.from_string(target_user_id)):
+            raise SynapseError(400, "Can only lookup local users")
```

Comment:
I think this was a copy and paste error?

```suggestion
            raise SynapseError(400, "Can only deactivate local users")
```